txpipe/covariance.py

```diff
@@ -658,7 +658,9 @@ def run(self):
         calculator = tjpcov.main.CovarianceCalculator({"tjpcov":tjp_config})
 
         cache = {'workspaces': workspaces}
-        covmat = calculator.get_all_cov_nmt(cache=cache)
+        tracer_nlcp = self.get_tracer_noise_from_sacc(cl_sacc)
+        covmat = calculator.get_all_cov_nmt(cache=cache,
```

Comment:
Will this function cope if the tracer_nlcp dictionary is empty?

Reply:
`tracer_nlp` should never be empty; i.e. should be always taken from what `TXTwoPointFourier` computes. In any case, I think it will not fail (but I think the expressions in TJPCov for the noise are different from those in TXPipe) because when passing the noise it checks first the tracer is in the dictionary: https://github.com/LSSTDESC/TJPCov/blob/01f6620d28d4c788eefbb1effd62139a22dcf3c8/tjpcov/main.py#L676-L677

Reply:
I've added a warning in case one of the tracers metadata `n_ell_coupled` is missing. I'm tempted to raise an error. I think it would be the safest option.

Reply:
Thanks! If there's never a reason that n_ell_coupled could be missing (e.g. are there any simple tests people might like to do where it's not there?) then an error makes sense, yep!

Reply:
At the moment, the code always computes the noise. So, it will always have `n_ell_coupled`. If this is modified later, one can add a flag or something. I've changed to an error.